pandas/core/_algos/__init__.py

```diff
@@ -0,0 +1,9 @@
+"""
+core._algos is for algorithms that operate on ndarray and ExtensionArray.
```

Comment:
Can we use `algos` instead of `_algos` here?

Reply:
I'm ambivalent on this, as I do like the non-private name, but we have a bunch of `import algorithms as algos` and i want to avoid ambiguity.

Reply:
then let’s just create core.algorithms as a subdir

Reply:
Hmm, that's a good point. The underscore is still only a small difference though. So maybe we should think of yet another name (or actually make algorithms.py into a module and rethink it more broader)

Reply:
So yes, similar as what @jreback said. 

But Brock, you wanted to better distinguish pure array algos, that don't need to deal with series/dataframe etc (which I think is a good goal). But then making algorithms.py into a submodule (and use that for the function in this PR) defeats that purpose?

Reply:
>  But then making algorithms.py into a submodule (and use that for the function in this PR) defeats that purpose?

That's my thought too.

> So maybe we should think of yet another name

Suggestions?